src/Microsoft.AspNet.HttpOverrides/HttpMethodOverrideMiddleware.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.HttpOverrides
+{
+    public class HttpMethodOverrideMiddleware
+    {
+        private const string xHttpMethodOverride = "X-Http-Method-Override";
+        private readonly RequestDelegate _next;
+        private readonly string _formFieldName;
+
+        public HttpMethodOverrideMiddleware(RequestDelegate next, string formFieldName = null)
+        {
+            _next = next;
+            _formFieldName = formFieldName;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (context.Request.Method == "POST")
+            {
+                if (_formFieldName != null && context.Request.HasFormContentType)
```

Comment:
These conditions make for some odd behaviors. Consider the following requests:
- _formFieldName is not set, Form POST, has override header, override header applied
- _formFieldName is set, Form POST without a matching _formFieldName, has override header, override header not applied